tests/VersionParserTest.php

```diff
@@ -401,7 +401,7 @@ public function caretConstraints()
      * @dataProvider hyphenConstraints
      *
      * @param string     $input
-     * @param Constraint $min
+     * @param Constraint|null $min
```

Comment:
I guess the `if ($min)` case in the unit test is dead-code because the dataprovider will never provide a  falsy min-value